Sources/Sentry/SentrySpan.m

```diff
@@ -84,10 +96,50 @@ - (instancetype)initWithContext:(SentrySpanContext *)context
         _spanId = context.spanId;
         _sampled = context.sampled;
         _origin = context.origin;
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        if (SentrySDK.options.enableContinuousProfiling) {
+            _profileSessionIDs = [NSMutableSet<NSString *> set];
+            if (SentryContinuousProfiler.isCurrentlyProfiling) {
+                [_profileSessionIDs
+                    addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+            }
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                addObserver:self
+                   selector:@selector(linkProfiler)
+                       name:kSentryNotificationContinuousProfileStarted];
+            [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+                addObserver:self
+                   selector:@selector(linkProfiler)
+                       name:kSentryNotificationContinuousProfileStopped];
+        }
+#endif // SENTRY_TARGET_PROFILING_SUPPORTED
     }
     return self;
 }
 
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+- (void)dealloc
+{
+    [self stopObservingContinuousProfiling];
+}
+
+- (void)linkProfiler
+{
+    [_profileSessionIDs addObject:SentryContinuousProfiler.currentProfilerID.sentryIdString];
+}
+
+- (void)stopObservingContinuousProfiling
+{
+    [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
```

Comment:
`m`: I think we should only call this if `enableContinuousProfiling` is enabled.